docs/fsharp/language-reference/compiler-directives.md

```diff
@@ -52,6 +52,31 @@ let str = "Debugging!"
 #endif
 ```
 
+## NULLABLE directive
+
+Starting with F# 9, you can enable nullable reference types in the project:
+
+```xml
+<Nullable>enable</Nullable>
+```
+
+This automatically sets `define:NULLABLE` directive to the build. It's useful while initially rolling out the feature, to conditionally change conflicting code by `#if NULLABLE` hash directives:
+
+```fsharp
+#if NULLABLE 
+let checkNonNull argName (arg: obj) =
+    match arg with
+    // nullness warning: the type 'obj' does not support 'null'
+    | null -> nullArg argName
```

Comment:
The if-else branches show identical code here.
Motivating example could e.g. be with a conditional `not null` constraint, or perharps for exposing a part of an public API.
Even the MaybeNull alias/wrapper could be used.

Reply:
Well, the idea was that the compiler directive makes the compiler produce a warning. We can have something more useful here - I added an example but not sure it counts really.